sklearn/utils/tests/test_class_weight.py

```diff
@@ -117,6 +117,15 @@ def test_compute_class_weight_balanced_unordered():
     assert_almost_equal(np.dot(cw, class_counts), y.shape[0])
     assert_array_almost_equal(cw, [2., 1., 2. / 3])
 
+def test_class_weight_with_string_label():
+    y = np.asarray(["A","A","A","B","B","C"])
+    classes = np.unique(y)
+    class_weights = {c: 1.0 for c in classes}
+    class_weights['D'] = 1.0  # This should get a proper ValueError
+    cw = assert_raises(ValueError, compute_class_weight, class_weights, 

```

Comment:
I used `test_compute_class_weight_not_present` as a template for the test I introduced, but did not want to change it.  I'm personally OK with multiple asserts in a single test case, but others (e.g., Robert Martin, "Clean Code", Ch 9) are against it, so I didn't want to upset others with multiple asserts.

I did incorporate your suggestion of using `assert_raise_message` which is an excellent idea - thank you.